upgrade-job/src/helm/client.rs

```diff
@@ -0,0 +1,154 @@
+use crate::{
+    common::error::{
+        Error::{HelmCommand, U8VectorToString, YamlParseFromSlice},
+        Result,
+    },
+    CliArgs,
+};
+use k8s_openapi::serde;
+use serde::Deserialize;
+use std::{process::Command, str};
+use tracing::info;
+
+#[derive(Clone, Deserialize)]
+pub(crate) struct HelmReleaseElement {
+    name: String,
+    //    namespace: String,
+    //    revision: String,
+    //    updated: String,
+    //    status: String,
+    chart: String,
+    //    app_version: String,
+}
+
+impl HelmReleaseElement {
+    pub(crate) fn name(&self) -> String {
+        self.name.clone()
+    }
+    pub(crate) fn chart(&self) -> String {
+        self.chart.clone()
+    }
+}
+
+pub(crate) struct HelmClient {
+    namespace: String,
+}
+
+impl HelmClient {
+    pub(crate) fn default(opts: &CliArgs) -> Self {
+        Self {
+            namespace: opts.namespace(),
+        }
+    }
+
+    pub(crate) fn list(
+        &self,
+        maybe_extra_args: Option<Vec<String>>,
+    ) -> Result<Vec<HelmReleaseElement>> {
+        let command: &str = "helm";
+        let mut args: Vec<String> = vec![
```

Comment:
why not use Vec<&str> and simply map to ToString later, so you don't need to have all the to_string all over the place.

Reply:
Using a macro to make it a bit more readable.